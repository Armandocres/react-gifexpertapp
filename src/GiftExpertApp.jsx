import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const initialState = ['One Punch'];

const GiftExpertApp = () => {

  const [categories, setCategories] = useState(initialState);

  return (
    <>
      <h1>GiftExpertApp</h1>
      <AddCategory setCategories={ setCategories } />
      <hr />
      <ol>
        { categories.map( category =>
          <GifGrid category={category} key={category} />
          ) }
      </ol>
    </>
  )
}

export default GiftExpertApp;