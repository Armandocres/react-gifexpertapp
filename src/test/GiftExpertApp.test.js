import React from 'react';
import GiftExpertApp from '../GiftExpertApp';
import { shallow } from 'enzyme';

describe('Pruebas en el componentes giftExpertApp', () => {

  test('Prueba del snapchot', () => {
    const wrapper = shallow(<GiftExpertApp />);
    expect(wrapper).toMatchSnapshot();
  })

  test('Debe mostrar la lista de categorias', () => {
    const categories = ['One Punch', 'Dragon Ball'];
    const warapper = shallow(<GiftExpertApp defaultCategories={categories} />);

    expect(warapper).toMatchSnapshot();
    expect(warapper.find('GifGrid').length).toBe(categories.length);

  })

})