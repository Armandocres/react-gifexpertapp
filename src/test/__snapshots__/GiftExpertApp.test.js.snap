// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componentes giftExpertApp Debe mostrar la lista de categorias 1`] = `
<Fragment>
  <h1>
    GiftExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en el componentes giftExpertApp Prueba del snapchot 1`] = `
<Fragment>
  <h1>
    GiftExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
